tests/ZfcRbacTest/Factory/ControllerGuardFactoryTest.php

```diff
@@ -50,13 +50,17 @@ public function testFactory()
             $this->getMock('ZfcRbac\Service\RoleService', [], [], '', false)
         );
 
-        $pluginManager = new GuardPluginManager();
-        $pluginManager->setServiceLocator($serviceManager);
+        $pluginManager = new GuardPluginManager($serviceManager);
 
         $factory         = new ControllerGuardFactory();
-        $controllerGuard = $factory->createService($pluginManager);
 
-        $this->assertInstanceOf('ZfcRbac\Guard\ControllerGuard', $controllerGuard);
-        $this->assertEquals(GuardInterface::POLICY_ALLOW, $controllerGuard->getProtectionPolicy());
+        if (method_exists($serviceManager, 'build')) { // servicemanager v3
```

Comment:
yep you are right but you also added test to test call to it. Makes sense? no :)